search-enterprise/latest/upgrade/index.md

```diff
@@ -43,3 +43,69 @@ Your current Content Services stack can continue to run while you are indexing t
 ![upgraded-environment]({% link search-enterprise/images/elasticsearch-upgrading-2.png %})
 
 > **Note:** You may need to use the Elasticsearch Re-indexing application to update to the latest changes. After that, new and updated documents will be uploaded to the Elasticsearch index by the Elasticsearch connector service using ActiveMQ messages.
+
+## Zero downtime upgrade
+
+You can upgrade from Search Services 2.x without experiencing any downtime, to Search Enterprise 3.1.1 when you are using Content Services 7.2 and above.
+
+1. Start an Elasticsearch 3.1.1 instance, for more see [Overview]({% link search-enterprise/latest/install/index.md %}).
+
+    Currently your installation currently only using solr.
+
+    ![add-empty]({% link search-enterprise/images/add-empty-elasticsearch.png %})
+
+2. Start a mirrored environment by replicating the content respository and Content Services.
+
+    You create a mirrored environment because the upgrade will not impact the primary environment. Use the Elasticsearch instance you created as the content repository for the mirrored environment. Once you have mirrored the environment do not change the content repository and only use it in read-only mode. If you do not need to preserve the content repository then you only need to mirror Content Services.
+
+    ![mirror-acs]({% link search-enterprise/images/mirror-acs-environment.png %})
+
+3. Create an Elasticsearch index by executing a search query on the mirrored environment.
+
+    Verify the index is created and its metadata correctly reflects your data model.
+
+    > **Note:** The index is not created when you mirrored the content repository and Content Services.
+
+4. Populate the index with existing data.
+
+    The index is populated and is based on the replicated database and is achieved by starting re-indexing on the mirrored environment. For more see [re-indexing]({% link search-enterprise/latest/config/index.md %}#alfresco-re-indexing-app).
+
+    **Note:** A window displays that states the primary database does not reflect the up to date index.
+
+    ![intial-reindexing]({% link search-enterprise/images/initial-re-indexing.png %})
+
+5. Shutdown the original environment.
+
+    You are left with an Elasticsearch server with a populated index.
+
+    **Note:** The index is not yet in sync with the primary environment.
+
+    ![shutdown-mirrored]({% link search-enterprise/images/shutdown-mirrored.png %})
+
+6. Keep your index up to date with changes made through Content Services.
+
+    To do this start live indexing on the primary environment. For more see [live-indexing]({% link search-enterprise/latest/config/index.md %}#alfresco-live-indexing-app).
+
+    > **Note:** Even after starting live indexing there is still a gap however between when you took a snapshot and when you started live indexing.
```

Comment:
Suggest "Even after starting live indexing there is still a gap from when you took a snapshot to when you started live indexing."

Reply:
Changed